shell/src/main/java/tachyon/shell/TFsShell.java

```diff
@@ -113,6 +113,53 @@ public int cat(TachyonURI path) throws IOException {
   }
 
   /**
+   * Copies a file or directory specified by argv from the under filesystem to the filesystem. 
+   *
+   * @param argv [] Array of arguments given by the user's input from the terminal
+   * @return 0 if command is successful, -1 if an error occurred.
+   * @throws IOException
+   */ 
+  public int load(TachyonURI filePath) throws IOException {
+    TachyonFS tachyonClient = createFS(filePath);
+    int ret = loadPath(tachyonClient, filePath);
+    if (ret == 0) {
+      System.out.println(filePath + " loaded");
+    } else {
+      System.out.println("Loading " + filePath + " failed");
+    }
+    return ret;
+  }
+
+  private int loadPath(TachyonFS tachyonClient, TachyonURI filePath) throws IOException {
+    TachyonFile tFile = tachyonClient.getFile(filePath);
+    Closer closer = Closer.create();
+    if (tFile == null) {
+      return -1;
+    }
+    if (tFile.isDirectory()) {
+      List<ClientFileInfo> files = tachyonClient.listStatus(filePath); 
+      Collections.sort(files);
+      for (ClientFileInfo file : files) {
+        TachyonURI newPath = new TachyonURI(file.getPath());
+        if (loadPath(tachyonClient, newPath) == -1) {
+          return -1;
+        }
+      }
+      return 0;
+    } else {
+      InStream in = closer.register(tFile.getInStream(ReadType.CACHE));
+      byte[] buf = new byte[8 * Constants.MB];
+      try {
+        while (in.read(buf) != -1) {
+        }
+        return 0;
+      } finally {
+        closer.close();
```

Comment:
initially thought so, but closer has another use of returning the right exception, which is kind of useful in this particular case. 
https://code.google.com/p/guava-libraries/wiki/ClosingResourcesExplained

added code to actually return the right exception.